### PR TITLE
Added testing and refactored list

### PR DESCRIPTION
**Pull request by dinodsaurus**
Connected to #5
Ok, I added testing using mocha/chai and also did some minor rewrite to list.js

### CODE REVIEW

**Comment by martinheidegger** [comment]
@dinodsaurus Awesome, lets get started. Uhm: Would you mind to separate the refactored list from the unit tests so we can talk only on one topic?

**Comment by dinodsaurus** [comment]
The problem is that the tests wont pass without this refactor(you have a lot of private methods, which are hard to test) but will keep it in mind for the future  :+1:

File: lib/standard/list.js [hunk 1]
```diff
@@ -4,6 +4,10 @@ var fs = require('fs')
 var path = require('path')
 var util = require('util')
 var ProgressBar = require('progress')
+var bb = require('bluebird')
+
+var json = require('../util/json')
+var validate = require('../util/validate')
```
Comment (martinheidegger) [inline lib/standard/list.js]: This file is required in the root of  `lib/standard/index.js` which in turn is required in the root `lib/index.js` to make sure that a simple `nodeschool-admin version` call doesn't load too many modules I put all the require statements on purse in the run method. The only "global" require statements are those generally required like `npmlog`, `path`. Other require statements should (for the sake of snappiness) preferably be loaded only when necessary. (Note: "ProgressBar" sneaked its way into the file header, shouldn't be there in my opinion)

**Comment by dinodsaurus** [comment]
Ok, sorry for this you have a good point, will move it back "in"

**Comment by dinodsaurus** [comment]
I think it is good idea that all the feature code follows this "functional" pattern

**Comment by martinheidegger** [comment]
@nodeschool/admin-maintainer how about we merge this now and continue to work on tests gradually?

**Comment by dinodsaurus** [comment]
This sounds good to me

**Comment by martinheidegger** [comment]
@fforres If you agree you can merge it and publish as 0.1.1 (for finger practice)

**Pull request merged**